core/src/test/java/org/apache/calcite/sql/type/RelDataTypeSystemTest.java

```diff
@@ -36,9 +43,13 @@
 class RelDataTypeSystemTest {
 
   /**
-   * Custom type system class that overrides the default decimal plus type derivation.
+   * Custom type system class that overrides the default decimal plus type derivation and
+   * overrides the max precision for timestamps.
    */
   private static final class CustomTypeSystem extends RelDataTypeSystemImpl {
+    // Arbitrarily choose a different maximum timestamp precision from the default.
+    private static final int CUSTOM_MAX_TIMESTAMP_PRECISION =
+        SqlTypeName.MAX_DATETIME_PRECISION + 3;
```

Comment:
I don't understand this, I'm curious why it is + `3`?

Reply:
I think he just chose an arbitrary number

Reply:
Yes, just the default offset by an arbitrary number (to ensure it is distinct from the default).